src/fr/perfscan/src/eu/kanade/tachiyomi/extension/fr/perfscan/PerfScan.kt

```diff
@@ -1,24 +1,136 @@
 package eu.kanade.tachiyomi.extension.fr.perfscan
 
-import eu.kanade.tachiyomi.multisrc.heancms.HeanCms
-import eu.kanade.tachiyomi.network.interceptor.rateLimitHost
-import okhttp3.HttpUrl.Companion.toHttpUrl
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import keiyoushi.utils.parseAs
+import okhttp3.Headers
 import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import java.text.DecimalFormat
+import java.text.SimpleDateFormat
+import java.util.Locale
 
-class PerfScan : HeanCms("Perf Scan", "https://perf-scan.fr", "fr") {
+class PerfScan : HttpSource() {
+    override val name = "Perf Scan"
+    override val baseUrl = "https://perf-scan.net"
+    private val apiUrl = "https://api.perf-scan.net"
+    override val lang = "fr"
+    override val supportsLatest = true
+    override val versionId = 2
 
-    override val client: OkHttpClient = super.client.newBuilder()
-        .rateLimitHost(apiUrl.toHttpUrl(), 1, 2)
-        .build()
+    override val client: OkHttpClient = network.cloudflareClient
 
-    init {
-        preferences.run {
-            if (contains("pref_url_map")) {
-                edit().remove("pref_url_map").apply()
+    private val dateFormat = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", Locale.US)
+    private val chapterNumberFormat = DecimalFormat("#.##")
+
+    override fun headersBuilder(): Headers.Builder = Headers.Builder()
```

Comment:
```suggestion
    override fun headersBuilder(): Headers.Builder = super.headerBuilder()
```
to include useragent